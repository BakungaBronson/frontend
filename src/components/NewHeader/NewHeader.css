header.NewHeader {
  background-color: #ffffff;
  color: #000000;
  display: flex;
  justify-content: space-between;
  height: 5rem;
  padding: 0 0 0 3rem;
  position: relative;
}

.HeaderLink {
  display: grid;
  grid-auto-flow: column;
  align-items: center;
  justify-content: center;
  overflow-y: hidden;
}

.HeaderLink a {
  font-size: 0.9em;
  color: #000;
  padding: 1rem 3rem;
  display: grid;
  align-content: center;
  height: 100%;
}

.HeaderLink a:hover{
  background-color: #F7B21F;
}

.HeaderLink .active{
  background-color: #F7B21F;
}
.LogoLink:focus {
  outline: none;
}
.LogoLink::-moz-focus-inner {
  border: 0;
}

.LoggedIn {
  display: flex;
}

.HeaderLinksWrap {
  display: flex;
}

.OnHead {
  display: grid;
  grid-auto-flow: column;
  align-items: center;
  justify-content: center;
  overflow-y: hidden;
}

.OnHead:hover .UserNames {
  background-color: #F7B21F;
  color: #000000;
}

.OnHead:hover svg {
  fill: #000;
}

div.DropDownGrid {
  display: flex;
  flex-direction: column;
}

div.HeaderLinksWrap.LoggedIn .BelowHeader {
  background-color: #f9f9f9;
  position: absolute;
  top: 100%;
  width: 10rem;
  right: 0;
}

.DropDownArrowSvg {
  width: 2rem;
  height: 2rem;
  fill: #000000;
}

.UserName {
  font-weight: bold;
  color: #000;
  cursor: default;
  -webkit-touch-callout: none; /* iOS Safari */
  -webkit-user-select: none; /* Safari */
  -khtml-user-select: none; /* Konqueror HTML */
  -moz-user-select: none; /* Old versions of Firefox */
  -ms-user-select: none; /* Internet Explorer/Edge */
  user-select: none; /* Non-prefixed version, currently
                                    supported by Chrome, Edge, Opera and Firefox */
}

.DropDownContent {
  color: #000;
  background-color: #f9f9f9;
  display: flex;
  flex-direction: column;
  z-index: 1;
  box-shadow: 0 4px 4px 0 rgba(117, 117, 117, 0.2);
  -webkit-touch-callout: none; /* iOS Safari */
  -webkit-user-select: none; /* Safari */
  -khtml-user-select: none; /* Konqueror HTML */
  -moz-user-select: none; /* Old versions of Firefox */
  -ms-user-select: none; /* Internet Explorer/Edge */
  user-select: none; /* Non-prefixed version, currently
                                    supported by Chrome, Edge, Opera and Firefox */
}

.DropDownLink {
  cursor: pointer;
  padding: 1rem 1.5rem;
}

.DropDownLink:hover,
.DropDownLink:active {
  background-color: #f1f1f1;
}

.DropDownLink, .DropDownLink:visited {
  color: #000;
  text-decoration: none;
}

.HeaderLinkBack {
  text-transform: uppercase;
  padding: 1rem 3rem;
  /* margin-left: 2rem; */
  font-size: 0.9em;
  padding: 1rem 3rem;
  height: 100%;
  display: grid;
  align-content: center;
}
.HeaderLinkBack:hover{
  background-color: #F7B21F;
}

.StripBorder {
  text-transform: uppercase;
  padding: 1rem 3rem;
  font-size: 0.9em;
  color: white;
  padding: 1rem;
}

a.HeaderLinkDocs{
  display: grid;
  align-content: center;
  padding: 1rem 3rem;
  height: 100%;
}

a.HeaderLinkDocs:hover{
  background-color: #F7B21F;
}

@media only screen and (max-width: 640px) {
  .HeaderLinkPricing,
  .HeaderLinkDocs {
    display: none;
  }
}
