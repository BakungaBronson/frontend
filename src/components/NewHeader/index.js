import React, { useState, useEffect, useRef } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import NewLogo from '../NewLogo';
import removeUser from '../../redux/actions/removeUser';
import './NewHeader.css';
import { DOCS_URL, BLOG_URL } from '../../config';

const NewHeader = (props) => {
  const [hidden, setHidden] = useState(false);
  const dropdownRef = useRef(null);
  const { user, match } = props;

  const toggleHidden = () => {
    if (hidden) {
      setHidden(false);
    } else {
      setHidden(true);
    }
  };

  const handleClickOutside = (event) => {
    if (dropdownRef.current && !dropdownRef.current.contains(event.target)) {
      setHidden(false);
    }
  };

  // componentWillMount & componentWillUnmount
  useEffect(() => {
    document.addEventListener('mousedown', handleClickOutside);

    // returned function will be called on component unmount
    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, []);

  return (
    <header className="NewHeader">
      <NewLogo />


      {(!user.accessToken || user.accessToken === '') && (
        <div className="HeaderLinksWrap">
          {match.path !== '/admin-login' && (
            <div className="HeaderLink bold">
              <a href={`${DOCS_URL}`} className="HeaderLinkDocs" rel="noopener noreferrer" target="_blank">Docs</a>
              <a href={`${BLOG_URL}`} className="HeaderLinkDocs" rel="noopener noreferrer" target="_blank">Blog</a>
              <Link to="/login" className="HeaderLinkDocs">Login</Link>
            </div>
          )}
        </div>
      )}

      {user.accessToken && (
        <div
          ref={dropdownRef}
          className="HeaderLink"
          onClick={toggleHidden}
          role="presentation"
        >
          {match.path === '/' || match.path === '/team' ? (
            <>
              <a href={`${DOCS_URL}`} className="HeaderLinkDocs" rel="noopener noreferrer" target="_blank">Docs</a>
              <a href={`${BLOG_URL}`} className="HeaderLinkDocs" rel="noopener noreferrer" target="_blank">Blog</a>
              <Link to={`/users/${user.data.id}/projects`} className="HeaderLinkDocs">Dashboard</Link>
            </>
          ) : (null)}

        </div>
      )}

    </header>
  );
};

NewHeader.propTypes = {
  removeUser: PropTypes.func.isRequired,
  user: PropTypes.shape({
    accessToken: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.bool
    ]),
    data: PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string
    }).isRequired,
  }),
  match: PropTypes.shape({
    path: PropTypes.string.isRequired
  }).isRequired,
};

NewHeader.defaultProps = {
  user: {}
};

const mapStateToProps = (state) => {
  const { user } = state;
  return { user };
};

const mapDispatchToProps = {
  removeUser
};

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(NewHeader));
