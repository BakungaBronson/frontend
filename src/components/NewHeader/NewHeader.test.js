import React from 'react';

import { shallow } from 'enzyme';

import NewHeader from "./";

const NewHeaderProps = {
  removeUser: jest.fn(),
  user: {
    data: {}
  },
  match: {
      path: ""
  }
};

describe('NewHeader component test', () => {
    const NewHeaderComponent = shallow(<NewHeader.WrappedComponent {...NewHeaderProps} />);
    it('checks the snapshot', () => {
        expect(NewHeaderComponent).toMatchSnapshot();
    })
})