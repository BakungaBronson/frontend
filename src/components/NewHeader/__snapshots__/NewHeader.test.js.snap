// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewHeader component test checks the snapshot 1`] = `ShallowWrapper {}`;
